controllers/argoutil/resource.go

```diff
@@ -33,6 +34,19 @@ import (
 	"github.com/argoproj-labs/argocd-operator/common"
 )
 
+const (
+	hashLabelLength = 7
+	// MaxLabelLength is the maximum length for Kubernetes labels and names
+	MaxLabelLength = 63
```

Comment:
Why `MaxLabelLength` has to have public access ? If all the truncation logic is within this utility file, we can make it private isn't it ? Can it be changed to `maxLabelLength` ? If this constant is required to be used else where then its better to make it a global constant in file `constants.go`.